cmake/flash/CMakeLists.txt

```
@@ -36,8 +36,17 @@ function(runners_yaml_append_config)
   get_runners_prop(elf_file elf "${KERNEL_ELF_NAME}")
   runners_yaml_append("  # Build outputs:")
   runners_yaml_append("  elf_file: ${elf}")
+
+  if(SYSBUILD)
+    zephyr_get(alt_file VAR ALT_SIGNED_HEX_FILE SYSBUILD)
+  endif()
+
   if(CONFIG_BUILD_OUTPUT_HEX)
-    get_runners_prop(hex_file hex "${KERNEL_HEX_NAME}")
+    if(SYSBUILD AND DEFINED alt_file)
```

you already tested for `if(SYSBUILD)` above, so if `SYSBUILD` is false then `alt_file` will be undefined.

So would be enough to just do:
```suggestion
    if(DEFINED alt_file)
```